File: src/swingjs/plaf/JSPanelUI.java
package swingjs.plaf;


import jsjava.awt.Dimension;

import jsjavax.swing.JRootPane;
import jsjavax.swing.LookAndFeel;

import swingjs.api.DOMNode;

public class JSPanelUI extends JSLightweightUI {

	int frameZ = 10000;
	private boolean isContentPane;
	public JSPanelUI() {
		isContainer = true;
		setDoc();
	}
	
	@Override
	public DOMNode createDOMNode() {
		if (domNode == null) {
			JRootPane root = jc.getRootPane();
			isContentPane = (jc == root.getContentPane());
			domNode = newDOMObject("div", id);
			if (isContentPane)
				DOMNode.setStyles(domNode, "overflow", "hidden");
			if (root != null && root.getGlassPane() == c)
				DOMNode.setStyles(domNode, "display", "none");
		}
    return domNode;
	}

	@Override
  protected Dimension setHTMLSize(DOMNode obj, boolean addCSS) {
		// SwingJS for now: just designated container width/height 
		return new Dimension(c.getWidth(), c.getHeight());
	}
	

	@Override
	protected void installUIImpl() {
    LookAndFeel.installColorsAndFont(jc,
        "Panel.background",
        "Panel.foreground",
        "Panel.font");
	}

	@Override
	protected void uninstallUIImpl() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public Dimension getPreferredSize() {
  	return null;
  }


}

File: site/swingjs/j2s/swingjs/plaf/JSPanelUI.js
Clazz.declarePackage ("swingjs.plaf");
Clazz.load (["swingjs.plaf.JSLightweightUI"], "swingjs.plaf.JSPanelUI", ["java.awt.Dimension", "javax.swing.LookAndFeel"], function () {
c$ = Clazz.decorateAsClass (function () {
this.frameZ = 10000;
this.isContentPane = false;
Clazz.instantialize (this, arguments);
}, swingjs.plaf, "JSPanelUI", swingjs.plaf.JSLightweightUI);
Clazz.makeConstructor (c$, 
function () {
Clazz.superConstructor (this, swingjs.plaf.JSPanelUI, []);
this.isContainer = true;
this.setDoc ();
});
Clazz.overrideMethod (c$, "getDOMObject", 
function () {
if (this.domNode == null) {
var root = this.c.getRootPane ();
this.isContentPane = (root != null && root.getContentPane () === this.c);
this.domNode = this.createDOMObject ("label", this.id, []);
}return this.domNode;
});
Clazz.overrideMethod (c$, "setHTMLSize", 
function (obj, addCSS) {
return  new java.awt.Dimension (this.c.getWidth (), this.c.getHeight ());
}, "swingjs.api.DOMNode,~B");
Clazz.defineMethod (c$, "getPreferredSize", 
function (c) {
return null;
}, "javax.swing.JComponent");
Clazz.overrideMethod (c$, "installJSUI", 
function () {
javax.swing.LookAndFeel.installColorsAndFont (this.c, "Panel.background", "Panel.foreground", "Panel.font");
});
Clazz.overrideMethod (c$, "uninstallJSUI", 
function () {
});
});
